tasks/connectors/edgescan/lib/edgescan_vulnerability.rb

```diff
@@ -50,7 +53,19 @@ def to_kenna_finding
         end
 
         def to_corresponding_kenna_asset
-          { "external_id" => external_asset_id, "tags" => asset.tags }
+          base = { "external_id" => external_asset_id, "tags" => asset.tags }
```

Comment:
@whitfieldsdad So what's not mentioned in the documentation there as far as I can see, is that in edgescan an `Asset` can have many `Location Specifiers`, which could be a `url`, `hostname`, `ip`, `cidr`, or `block`. What the edgescan connector is doing is translating each `Location Specifier` into a Kenna asset.

Sometimes a vulnerability is found in edgescan which doesn't have an exact matching  `Location Specifier`. In this case a Kenna asset will be created, using the location where the vulnerability was found, so the vulnerability can be linked to it.

Using `Hosts` wouldn't capture everything from edgescan's side either. Not every edgescan `Asset` or `Location Specifier` has a matching host. The `Host` information is auto-generated by the ASM scans we do. Where as an `Asset` is added by a person.

Reply:
That makes sense, thanks!

I think that the best bet for simplifying the logic here would be to avoid using the strategy pattern and instead have one only parser per output object type (i.e. one function for creating Kenna assets, one for creating vulnerabilities, and one for creating findings).

You could then pass in any additional information if it's available (e.g. hostnames from host objects).

I don't know Ruby very well, but in Python, you could do something like this:

```python
def parse_host(asset: dict, vulnerability: dict, host: Optional[dict] = None) -> dict:
  hostname = None
  if host:    
    hostnames = host['hostnames']
    if hostnames:
      hostname = hostnames[0]

return {
    'ip_address': asset['locator'],
    'tags': asset['tags'],
    'hostname': hostname
  }
```

Reply:
AFAIK this change should basically cut your code in half

Reply:
Fixed in [55031b4](https://github.com/KennaSecurity/toolkit/pull/409/commits/55031b4c31b0f90f022e2f7dd3bdd00c048ba55b)